File: dp_labs_and_solutions/original/dp_labs_and_solutions/dp_matlab_labs_and_solutions/decorator/lab/CoffeeWithoutMilkWithSugar.m
% CoffeeWithoutMilkWithSugar.m

classdef CoffeeWithoutMilkWithSugar < CoffeeWithoutMilk
	methods
		function prepare(theCoffeeWithoutMilkWithSugar)
			prepare@CoffeeWithoutMilk(theCoffeeWithoutMilkWithSugar);
			fprintf('Adding sugar.\n');
		end
	end
end

File: dp_labs_and_solutions/original/dp_labs_and_solutions/dp_cpp_labs_and_solutions/decorator/lab/coffee_without_milk_with_sugar.cpp
// coffee_without_milk_with_sugar.cpp

#include <iostream>
#include "coffee_without_milk.h"
#include "coffee_without_milk_with_sugar.h"

void coffee_without_milk_with_sugar::prepare()
{
	coffee_without_milk::prepare();
	std::cout << "Adding sugar.\n";
}
